### PR TITLE
Feat: Added cartPage+refactoring

### PR DESCRIPTION
**Pull request by RahulSingh9131**
This branch contains
1.some basic refactoring
2.cart page

### CODE REVIEW

**Review by shuntzhere — APPROVED** [review]
The code looks fine, liked the use of react toastify. Don't see any commented code either. Nice work.

File: src/components/ProductCard.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import { ToastContainer, toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+import { useCart } from '../context/CartContext';
+
+function ProductCard({_id,categoryName,src,title,originalPrice,price,discount,rating,quantity}) {
+
+    const {cartState,cartDispatch}=useCart();
+
+    const {cartBasket}=cartState;
+    console.log(cartBasket);
```
Comment (nishasen) [inline src/components/ProductCard.js]: remove the console.logs

File: src/components/ProductCard.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            item:{
+                _id:_id,
+                categoryName:categoryName,
+                src:src,
+                title:title,
+                originalPrice:originalPrice,
+                price:price,
+                discount:discount,
+                rating:rating,
+                quantity:quantity
```
Comment (nishasen) [inline src/components/ProductCard.js]: You can use object spread here

File: src/css/screens/cart.css [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+}
+.cartpage .number-rating-container{
+    margin-top: var(--mb-0-5);
+    width: 100%;
+    max-width: 5rem;
+}
+.cartpage .quantity-btn{
+    display: flex;
+    align-items: center;
+    justify-content: center;
```
Comment (nishasen) [inline src/css/screens/cart.css]: use util class here

**Review by nishasen — APPROVED** [review]
Your code looks good , have suggested small things.

**Pull request merged**